src/App.tsx

```diff
@@ -1,223 +1,134 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+
+import {
+  setIsLoadingContext,
+  functonsContext,
+} from './components/Context/context';
+import { Todo } from './types/Todo';
+import { Filter } from './components/Filter/Filter';
+import { NewTodo } from './components/NewTodo/NewTodo';
+import { TodoList } from './components/TodoList/TodoList';
+import { getTodos } from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import {
+  ErrorNotification,
+} from './components/ErrorNotification/ErrorNotification';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [allTodos, setAllTodos] = useState<Todo[] | null>(null);
+  const [isLoading, setIsLoading] = useState<number[]>([]);
+  const [activeTodos, setActiveTodos] = useState<Todo[] | null>(null);
+  const [errorStatus, setErrorStatus] = useState('');
+  const [currentInput, setCurrentInput] = useState('');
+  const [filter, setFilter] = useState('all');
+
+  const visibleTodos = useMemo(() => {
+    let result = allTodos;
```

Comment:
second, don't use let and ifs, better use switch/case and return values right away